common/workunit/workunit.cpp

```diff
@@ -10013,6 +10416,93 @@ IWUGraphStats *CLocalWorkUnit::updateStats(const char *graphName, StatisticCreat
     return new CWuGraphStats(LINK(p), creatorType, creator, _wfid, graphName, subgraph);
 }
 
+bool CLocalWorkUnit::usingDedicatedLogFiles() const
+{
+    if (hasDebugValue("imported"))
```

Comment:
ok, what will currently happen (as this PR stands) with those other places, if the workunit originally came from a zap report?

Reply:
If the workunit originally came from a zap report, its XML has the Debug  Value "imported". Both thor log and an eclagent log can be retrieved by a log file name. To use @singlelog, the thor or eclagent process name has to be passed around.

Reply:
in fact it's not clear why usingDedicatedLogFiles() (or Debug/@imported) is needed at all.
It is not called anywhere afaics.

Now that the individual Process/ entries direct any code that reads the logs whether it is @singleLog or not.

However, I see code that doesn't call usingDedicatedLogFiles() but calls hasDebugValue("imported") directly, i.e.:
```
    if (hasDebugValue("imported")) //WU imported from a ZAP report
        thorMasterLogDateSearchString.set("_thormaster.");
    else
        thorMasterLogDateSearchString.set("/thormaster.");
```

that should be able to deduce this from the IConstWUThorLogInfo information, without need for specifically looking for  "improted".

Reply:
Let me confirm: 1. the singlelog attribute inside the Process/ entries (not the "imported") should be used to decide single log file or not.  2. the IConstWUThorLogInfo information (the pattern attribute) should be used to build the thorMasterLogDateSearchString without checking the "imported".

Reply:
yes I think so, can you foresee a problem with that approach?

That way there is no special handling/conditinal code based on whether it is imported, only differential handling (handled by the IConstWUThorLogInfo impl.) based on whether single log or not.
Which we will want anyway at some point even for standard (non imported) workunits.

Reply:
I do not foresee any problem for the 1st item (just some code changes as I mentioned before). For the second item,  instead of using the thorMasterLogDateSearchString, I may use the pathTail on the log attribute (not the pattern attribute) from the IConstWUThorLogInfo information.

Reply:
> I may use the pathTail on the log attribute (not the pattern attribute) from the IConstWUThorLogInfo information.

yes, could be done within CWUThorLogInfo ctor?

Reply:
I may enhance the IConstWUThorLogInfo  to IConstWUProcessLogInfo because similar functions may be used for eclagent log.

Reply:
That sounds sensible, hopefully the outcome will come simpler.